docs/book/v3/events.md

```diff
@@ -0,0 +1,251 @@
+use Swoole\Http\Request as SwooleHttpRequest;
+use Swoole\Http\Response as SwooleHttpResponse;
+use Swoole\Http\Server as SwooleHttpServer;
+
+/**
+ * Describes "start" event
+ */
+class ServerStartEvent
+{
+    public function getServer(): SwooleHttpServer;
+}
+
+/**
+ * Describes "managerstart" event
+ */
+class ManagerStartEvent
+{
+    public function getServer(): SwooleHttpServer;
+}
+
+/**
+ * Describes "managerstop" event
+ */
+class ManagerStopEvent
+{
+    public function getServer(): SwooleHttpServer;
+}
+
+/**
+ * Describes "workerstart" event
+ */
+class WorkerStartEvent
+{
+    public function getServer(): SwooleHttpServer;
+    public function getWorkerId(): int;
+}
+
+/**
+ * Describes "workerstop" event
+ */
+class WorkerStopEvent
+{
+    public function getServer(): SwooleHttpServer;
+    public function getWorkerId(): int;
+}
+
+/**
+ * Describes "workererror" event
+ */
+class WorkerErrorEvent
+{
+    public function getServer(): SwooleHttpServer;
+    public function getWorkerId(): int;
+    public function getExitCode(): int;
+    public function getSignal(): int;
+}
+
+/**
+ * Describes "request" event
+ */
+class RequestEvent implements StoppableEventInterface
+{
+    public function isPropagationStopped(): bool;
+    public function getRequest(): SwooleHttpRequest;
+    public function getResponse(): SwooleHttpResponse;
+    public function responseSent(): void;
+}
+
+/**
+ * Describes "beforereload" event (prior to hot code reload)
+ */
+class BeforeReloadEvent
+{
+    public function getServer(): SwooleHttpServer;
+}
+
+/**
+ * Describes "afterreload" event (after hot code reload)
+ */
+class AfterReloadEvent
+{
+    public function getServer(): SwooleHttpServer;
+}
+
+/**
+ * Describes "task" event
+ */
+class TaskEvent implements StoppableEventInterface
+{
+    public function isPropagationStopped(): bool;
+    public function getServer(): SwooleHttpServer;
+    public function getTaskId(): int;
+    public function getWorkerId(): int;
+
+    /**
+     * Data passed to the task() method
+     */
+    public function getData(): mixed;
+
+    /**
+     * "Return value" of the task; what to pass to finish()
+     *
+     * @param mixed $returnValue
+     */
+    public function setReturnValue($returnValue): void;
+
+    /** @return mixed */
+    public function getReturnValue();
+
+    /**
+     * Call this to indicate processing is complete, and to stop processing
+     * further listeners.
+     */
+    public function taskProcessingComplete(): void;
+}
+
+/**
+ * Describes "finish" event, called when processing a task has been completed
+ */
+class TaskFinishEvent
+{
+    public function getServer(): SwooleHttpServer;
+    public function getTaskId(): int;
+
+    /**
+     * Data passed to the finish() method
+     */
+    public function getData(): mixed;
+}
+
+
+/**
+ * Describes "shutdown" event
+ */
+class ServerShutdownEvent
+{
+    public function getServer(): SwooleHttpServer;
+}
+```
```

Comment:
I would suggest splitting this code block. Each event is listed with a headline and a code block. For example:

``````
### Shutdown Event

```php
namespace Mezzio\Swoole\Event;

use Swoole\Http\Server;

class ServerShutdownEvent
{
    public function getServer(): Server;
}
```